### PR TITLE
Rename top-level module to `circuit_knitting`

### LINKED ISSUE #221: Shorten top-level module names
We've talked before about shortening some of the import names.  I propose changing

```python
from circuit_knitting_toolbox.entanglement_forging import EntanglementForgingAnsatz
from circuit_knitting_toolbox.circuit_cutting import partition_circuit_qubits
```

to

```python
from circuit_knitting.forging import EntanglementForgingAnsatz
from circuit_knitting.cutting import partition_circuit_qubits
```

#### Rationale

- `circuit_knitting` is descriptive enough.  `_toolbox` does not need to be mentioned in every import location.  Previously, we've considered `ckt`, but this is non-descriptive and could potentially clash with another package (the three letter namespace is typically very overloaded)
- `circuit_knitting.circuit_cutting` is redundant.  `circuit_knitting.cutting` is less of a mouthful
- I'm more torn between `circuit_knitting.entanglement_forging` and `circuit_knitting.forging`; however, if somebody is using this module, they probably have a good idea of what "forging" is about.

#### Other notes

Let's keep the documentation directories the way they are rather than replace `circuit_cutting` and `entanglement_forging` with shorter names.  This will
1. prevent links from unnecessarily breaking
2. provide more descriptive names to people navigating the documentation, e.g., the tutorials

### PR DESCRIPTION
**Pull request by garrison**
This performs the renames porposed in #221 and thus closes #221.

In order to demonstrate that the deprecated import locations work, I have not updated the imports in the tests or notebooks yet.  I'll do that once CI passes.

I also made it so docs will no longer deploy for pushes to `main`, since `main` will now be ahead of the current stable release.

### CODE REVIEW

File: circuit_knitting/__init__.py
Comment (garrison) [inline circuit_knitting/__init__.py:1]: This PR needs a release note. I will add one in the morning.

File: pyproject.toml [hunk 1]
```diff
@@ -93,12 +93,18 @@ notebook-dependencies = [
 "Documentation" = "https://qiskit-extensions.github.io/circuit-knitting-toolbox/"
 "Repository" = "https://github.com/Qiskit-Extensions/circuit-knitting-toolbox"
 
+[tool.hatch.build.targets.wheel]
+only-include = [
+    "circuit_knitting",
```
Comment (caleb-johnson) [inline pyproject.toml:98]: So users will have access to two namespaces, and one of the namespaces will throw a bunch of warnings?

**Review by caleb-johnson — APPROVED** [review]
This looks really great, nice job. I had a couple questions out of interest but lgtm

File: circuit_knitting_toolbox/circuit_cutting/__init__.py [hunk 2]
```diff
@@ -90,3 +39,35 @@
     "PartitionedCuttingProblem",
     "CuttingExperimentResults",
 ]
+
+sys.modules["circuit_knitting_toolbox.circuit_cutting.qpd"] = qpd
```
[the remark this answers is not in the document]
Comment (garrison) [reply circuit_knitting_toolbox/circuit_cutting/__init__.py:43]: It tells the module system that if somebody tries to import a module called `circuit_knitting_toolbox.circuit_cutting.qpd`, then instead use this other thing that's already been loaded, which in this case is `circuit_knitting.cutting.qpd`.

This is explained at bit at https://stackoverflow.com/a/72244240/1558890.  (I tried the lazy method mentioned there too, but it had more issues than was worth.)

File: pyproject.toml (hunk 1 above)
Comment (garrison) [reply pyproject.toml:98]: Exactly.

**Comment by garrison** [comment]
> One more thing, should we rename the docs directories to `cutting` and `forging` ?

This was an intentional choice, but we could re-evaluate.  Quoting from #221:

> Let's keep the documentation directories the way they are rather than replace circuit_cutting and entanglement_forging with shorter names. This will
>
> - prevent links from unnecessarily breaking
> - provide more descriptive names to people navigating the documentation, e.g., the tutorials

**Comment by caleb-johnson** [comment]
> > One more thing, should we rename the docs directories to `cutting` and `forging` ?
> 
> This was an intentional choice, but we could re-evaluate. Quoting from #221:
> 
> > Let's keep the documentation directories the way they are rather than replace circuit_cutting and entanglement_forging with shorter names. This will
> > 
> > * prevent links from unnecessarily breaking
> > * provide more descriptive names to people navigating the documentation, e.g., the tutorials

I agree with both. I don't think this is a source of confusion

File: releasenotes/notes/rename-top-level-import-bad074e7a8f348c9.yaml [hunk 3]
```diff
@@ -0,0 +1,14 @@
+---
+upgrade:
+  - |
+    The top-level name for imports has been renamed from
```
Comment (caleb-johnson) [inline releasenotes/notes/rename-top-level-import-bad074e7a8f348c9.yaml:4]: Aren't deprecations normally replaced with a new target or a new way to migrate? How is this different? Isn't it a deprecation?

**Review by caleb-johnson — APPROVED** [review]

**Pull request merged**